Look into `AnalyzerTests.test_getManifests` failure

```
2020-05-26T16:36:10.9671952Z /__w/SwiftPackageIndex-Server/SwiftPackageIndex-Server/Tests/AppTests/
AnalyzerTests.swift:347: error: AnalyzerTests.test_getManifests : XCTAssertEqual failed: ("[]") is not equal to ("["git checkout \"0.4.2\" --quiet", "swift package dump-package"]") - 
```

https://github.com/SwiftPackageIndex/SwiftPackageIndex-Server/runs/710087848

https://pipelines.actions.githubusercontent.com/8gSpUYYn2yLiXm9OtIrgygCWaoz1YKaNeE87jtNyKJiwoAqGuC/_apis/pipelines/1/runs/581/signedlogcontent/3?urlExpires=2020-05-26T16%3A41%3A13.3000719Z&urlSigningMethod=HMACV1&urlSignature=%2F2jW2Jps%2Bsjqz2aCY2bj6zvqfTnIx1o2oT7iSkSrb3w%3D

--- a/Tests/AppTests/AnalyzerTests.swift
+++ b/Tests/AppTests/AnalyzerTests.swift
@@ -25,7 +25,7 @@ class AnalyzerTests: AppTestCase {
         let queue = DispatchQueue(label: "serial")
         var commands = [Command]()
         Current.shell.run = { cmd, path in
-            queue.async {
+            queue.sync {
                 let c = cmd.string.replacingOccurrences(of: checkoutDir!, with: "...")
                 let p = path.replacingOccurrences(of: checkoutDir!, with: "...")
                 commands.append(.init(command: c, path: p))
@@ -56,7 +56,7 @@ class AnalyzerTests: AppTestCase {
         let outDir = try XCTUnwrap(checkoutDir)
         XCTAssert(outDir.hasSuffix("SPI-checkouts"), "unexpected checkout dir, was: \(outDir)")
         XCTAssertEqual(commands.count, 30)
-        assertSnapshot(matching: Set(commands), as: .dump)
+        assertSnapshot(matching: commands, as: .dump)
 
         // validate versions
         // A bit awkward... create a helper? There has to be a better way?
@@ -142,7 +142,7 @@ class AnalyzerTests: AppTestCase {
         let queue = DispatchQueue(label: "serial")
         var commands = [Command]()
         Current.shell.run = { cmd, path in
-            queue.async {
+            queue.sync {
                 commands.append(.init(command: cmd.string, path: path))
             }
             if cmd.string == "git tag" {
@@ -297,7 +297,7 @@ class AnalyzerTests: AppTestCase {
         let queue = DispatchQueue(label: "serial")
         var commands = [String]()
         Current.shell.run = { cmd, _ in
-            queue.async {
+            queue.sync {
                 commands.append(cmd.string)
             }
             if cmd.string == "swift package dump-package" {
@@ -326,7 +326,7 @@ class AnalyzerTests: AppTestCase {
         let queue = DispatchQueue(label: "serial")
         var commands = [String]()
         Current.shell.run = { cmd, _ in
-            queue.async {
+            queue.sync {
                 commands.append(cmd.string)
             }
             if cmd.string == "swift package dump-package" {
@@ -498,7 +498,7 @@ class AnalyzerTests: AppTestCase {
         let queue = DispatchQueue(label: "serial")
         var commands = [String]()
         Current.shell.run = { cmd, path in
-            queue.async {
+            queue.sync {
                 let c = cmd.string.replacingOccurrences(of: checkoutDir, with: "...")
                 commands.append(c)
             }

--- a/Tests/AppTests/AnalyzerTests.swift
+++ b/Tests/AppTests/AnalyzerTests.swift
@@ -56,7 +56,9 @@ class AnalyzerTests: AppTestCase {
         let outDir = try XCTUnwrap(checkoutDir)
         XCTAssert(outDir.hasSuffix("SPI-checkouts"), "unexpected checkout dir, was: \(outDir)")
         XCTAssertEqual(commands.count, 30)
-        assertSnapshot(matching: commands, as: .dump)
+        // We need to sort the issued commands, because macOS and Linux have stable but different
+        // sort orders o.O
+        assertSnapshot(matching: commands.sorted(), as: .dump)
 
         // validate versions
         // A bit awkward... create a helper? There has to be a better way?
@@ -515,9 +517,14 @@ class AnalyzerTests: AppTestCase {
 }
 
 
-struct Command: Equatable, CustomStringConvertible, Hashable {
+struct Command: Equatable, CustomStringConvertible, Hashable, Comparable {
     var command: String
     var path: String
 
     var description: String { "'\(command)' at path: '\(path)'" }
+
+    static func < (lhs: Command, rhs: Command) -> Bool {
+        if lhs.command < rhs.command { return true }
+        return lhs.path < rhs.path
+    }
 }

--- a/Tests/AppTests/Util.swift
+++ b/Tests/AppTests/Util.swift
@@ -13,6 +13,7 @@ func setup(_ environment: Environment, resetDb: Bool = true) throws -> Applicati
     Current = .mock
     
     let app = Application(.testing)
+    app.logger.logLevel = Environment.get("LOG_LEVEL").flatMap(Logger.Level.init(rawValue:)) ?? .warning
     try configure(app)
     if resetDb {
         try app.autoRevert().wait()

